pkg/dogstatsd/server.go

```diff
@@ -547,11 +569,49 @@ func (s *Server) errLog(format string, params ...interface{}) {
 	}
 }
 
-func (s *Server) parseMetricMessage(metricSamples []metrics.MetricSample, parser *parser, message []byte, origin string) ([]metrics.MetricSample, error) {
+// createOriginTagMaps  keeps these in a cache to avoid a lot of heap escape
+// that we can't avoid in this hot path
+func (s *Server) createOriginTagMaps(origin string) cachedTagsOriginMap {
```

Comment:
Could we benchmark the impact of this change. This is a very hot path for dsd.

Reply:
This feature is behind the DogStatsD debug flag, which means the performance impact will only occur on instances where we explicitly switched DogStatsD to debug mode, knowing that performances will decrease. Main idea behind this new feature is to ease troubleshooting on nodes where we already identified there is an issue.
I agree that it would be nice to know the performance impact on this addition but I feel that we don't really need to.
I'll try to gather that anyway.

Reply:
I think knowing if it's usable on prod would be nice.

Reply:
If it's for troubleshooting only we could just accumulate the contexts without ever cleaning them, no ?

Reply:
In all honesty, the rest of this debug feature is already way more CPU/mem consuming than this addition, so I would not mind too much about this.
I think that 200 contexts is already large enough for multiple hours of run with this feature, don't you?

Reply:
I think it is. I just spent quite some time troubleshooting DSD and realized that adding more latency to the pipeline makes it harder to troubleshoot has it blur the performances. Like adding Prometheus summaries added a lot a packet drop but not proportionally to the base line: scenario that were not dropping started dropping a lot and scenario that were already dropping a lot of packet only increase slightly.

I just want us to be sure we can use this on production without doubling packet drops. this could be an easy use case og the benchmark env, WDYT ?

Reply:
We observed a 6% CPU increase in the benchmarks.